templates/JSResolverOCHTTPSTemplate.js

```diff
@@ -394,16 +398,26 @@ function createQueryFunctionMatchStatement(obj, matchStatements, querySchemaInfo
             gq = gq.replace('$' + arg.name.value, arg.value.value);
         });
                 
-        matchStatements.push(gq);                
+        matchStatements.push(gq);
             
     } else {
 
         let { queryArguments, where } = getQueryArguments(obj.definitions[0].selectionSet.selections[0].arguments, querySchemaInfo);
         
         if  (queryArguments.length > 0) {
-            matchStatements.push(`MATCH (${querySchemaInfo.pathName}:${querySchemaInfo.returnTypeAlias}{${queryArguments}})${where}`);
+            if (colonCount(querySchemaInfo.returnTypeAlias) > 0) {
```

Comment:
You don't need to determine if there are colons first, just always back tick the node label.